File: src/ia_dev_env/assembler/conditions.py
from __future__ import annotations

from typing import List

from ia_dev_env.models import ProjectConfig


def extract_interface_types(config: ProjectConfig) -> List[str]:
    """Return list of interface type strings from config."""
    return [iface.type for iface in config.interfaces]


def has_interface(config: ProjectConfig, iface_type: str) -> bool:
    """Check if a specific interface type exists in config."""
    return any(iface.type == iface_type for iface in config.interfaces)


def has_any_interface(config: ProjectConfig, *types: str) -> bool:
    """Check if any of the specified interface types exist."""
    type_set = set(types)
    return any(iface.type in type_set for iface in config.interfaces)

File: src/assembler/conditions.ts
import type { ProjectConfig } from "../models.js";

/**
 * Return list of interface type strings from config.
 */
export function extractInterfaceTypes(
  config: ProjectConfig,
): string[] {
  return config.interfaces.map((iface) => iface.type);
}

/**
 * Check if a specific interface type exists in config.
 */
export function hasInterface(
  config: ProjectConfig,
  ifaceType: string,
): boolean {
  return config.interfaces.some(
    (iface) => iface.type === ifaceType,
  );
}

/**
 * Check if any of the specified interface types exist.
 */
export function hasAnyInterface(
  config: ProjectConfig,
  ...types: string[]
): boolean {
  const typeSet = new Set(types);
  return config.interfaces.some(
    (iface) => typeSet.has(iface.type),
  );
}
